src/Executables/Game/World/Map.cs

```diff
@@ -295,12 +296,17 @@ private MonsterEntity SpawnMonster(uint id, SpawnPoint spawnPoint)
                 0
             );
 
-            if (((EAiFlags)monster.Proto.AiFlag).HasAnyFlags(EAiFlags.NoMove))
+            if (monster.Proto.AiFlag.HasAnyFlags(EAiFlags.NoMove))
             {
                 monster.PositionX = (int)(PositionX + baseX * SPAWN_POSITION_MULTIPLIER);
                 monster.PositionY = (int)(PositionY + baseY * SPAWN_POSITION_MULTIPLIER);
                 var compassDirection = (int)spawnPoint.Direction - 1;
-                var rotation = 45 * (compassDirection < 0 ? 8 : compassDirection);
+                
+                if (compassDirection < 0)
+                {
+                    compassDirection = (int)ESpawnPointDirection.Random;
```

Comment:
Shouldn't random mean that it picks a random direction? It's current `0 * 45` but should be `Random(0, 8) * 45`?

Reply:
Also shouldn't it be `== 0` not `< 0` ?
What about values bigger than max? (May never happen but I would interpret it as random)

Reply:
At line 303, we subtract 1 from the `spawnPoint.Direction` because `monster.Rotation` is in degrees and true south of map is 0 degrees. In order to achieve true south we need to `45 * (spawnPoint.Direction - 1)` . Our formula would be wrong about 45 degrees for south if  `45 * spawnPoint.Direction`

Also between lines 318 and 321, we create a random rotation. Spawn group data has lots of `0` directions and I guess we don't want to spawn monsters facing a certain direction :D 

I'll add an overflow protection as you would 👍🏻

Reply:
At line 303, we subtract 1 from `spawnPoint.Direction` because `monster.Rotation` is in degrees, and true south on the map is 0 degrees. To achieve true south, we need to use `45 * (spawnPoint.Direction - 1)`. Our formula would be off by 45 degrees for south if we just used `45 * spawnPoint.Direction`.

Also, between lines 318 and 321, we create a random rotation for covering the case above. The spawn group data has lots of 0 directions, and I guess we don't want to spawn monsters facing the same direction. :D

I'll add overflow protection as you suggested. 👍🏻

Reply:
lgtm